editor/utils/editor-can-upload/index.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Response cache of response header.
+ *
+ * @type {Object}
+ */
+export const cachedMediaHeader = {};
+
+/**
+ * Whether a request is running.
+ *
+ * @type {Boolean}
+ */
+let isRequestingMediaOptions = false;
+
+export function requestMediaAllowHeader( request ) {
+	isRequestingMediaOptions = true;
+
+	const promise = wp.apiRequest( request )
+		.then( ( body, status, xhr ) => {
+			return xhr.getResponseHeader( 'allow' );
+		} )
+		.then( ( response ) => {
+			cachedMediaHeader.allow = response;
+			isRequestingMediaOptions = false;
+		} );
+
+	return Promise.resolve( promise );
+}
+
+export function getMediaAllowHeader() {
+	const request = { path: '/wp/v2/media', method: 'OPTIONS' };
+
+	if ( cachedMediaHeader.allow ) {
+		return cachedMediaHeader.allow;
+	}
+
+	return requestMediaAllowHeader( request );
+}
+
+export default function editorCanUpload() {
+	if ( isRequestingMediaOptions ) {
+		return;
+	}
+
+	const allowMedia = getMediaAllowHeader();
+
+	return 'GET, POST' === allowMedia;
```

Comment:
What's to say this isn't returned as `POST, GET` ? Or `GET,POST` ? Or `GET, POST, DELETE` ? (Now or in the future)

Should we consider these as individual verbs, where we're chiefly concerned with the presence of `POST` ?

e.g.

```js
return includes( allowMedia.split( ',' ).map( ( verb ) => verb.trim() ), 'POST' );
```

https://lodash.com/docs/4.17.10#includes

Should we just cache this boolean value, rather than the header object?